src/auth/use-subscribe-to-auth-user-change.ts

```diff
@@ -11,25 +12,36 @@ export const useSubscribeToAuthUserChange = (
   const [resubscribeToggle, setResubscribeToggle] = useState(true);
 
   useEffect(() => {
+    Bugsnag.leaveBreadcrumb('Subscribing to auth user changes');
     dispatch({type: 'SET_AUTH_STATUS', authStatus: 'loading'});
+    let signInInitiated = false;
     const unsubscribe = auth().onUserChanged((user) => {
       if (user) {
         updateMetadata({
-          'AtB-Firebase-Auth-Id': user.uid,
-          'AtB-Auth-Type': getAuthenticationType(user),
+          'AtB-Firebase-Auth-Id': user?.uid,
+          'AtB-Auth-Type': mapAuthenticationType(user),
         });
-        dispatch({type: 'SET_USER', user});
-      } else {
+        dispatch({
+          type: 'SET_USER',
+          userId: user?.uid,
+          phoneNumber: user?.phoneNumber || undefined,
+          authenticationType: mapAuthenticationType(user),
+        });
+      } else if (!signInInitiated) {
         /*
         Sign in anonymously if the onUserChanged event fired immediately on
         subscription did not include user data. In other words, user was not
         previously signed in.
          */
-        auth().signInAnonymously();
+        auth()
+          .signInAnonymously()
+          .then(() => {
+            signInInitiated = true;
```

Comment:
The `signInAnonymously` just starts the process, but the actual signing in happens async. The reason the function returns a promise is because it is a network request to Firebase. If it was implemented like this:
```ts
signInInitiated = true;
auth().signInAnonymously()
```

The the `signInInitiated` value would be set to true even if `signInAnonymously` failed, for example because of missing network connection.

Reply:
Right! A bit confusing with a `.then` on a call like that happening before the process is completed 😅